### PR TITLE
`gpi-waiting-list.php`: Refactor to class and add support for Single Products

### PR DESCRIPTION
**Pull request by claygriffiths**
## Context

⛑️ Ticket(s): https://secure.helpscout.net/conversation/1972333442/37514/

## Summary

* Refactor to class
* Add support for waiting list on Single Products (previously only choice-based)

### CODE REVIEW

File: gp-inventory/gpi-waiting-list.php [hunk 1]
```diff
@@ -8,99 +8,197 @@
...
+		 */
+		add_filter( 'gpi_disable_choices', '__return_false' );
+		add_filter( 'gpi_remove_choices', '__return_false' );
+
+		add_filter( 'gpi_pre_render_choice', array( $this, 'pre_render_choice' ), 10, 5 );
+
+		// Add support for showing waiting list message in confirmations and notifications.
+		add_filter( 'gform_pre_submission_filter', array(
+			$this,
+			'add_waitlist_message_to_choices_on_submission'
```
Comment (spivurno) [inline gp-inventory/gpi-waiting-list.php:32]: I'm assuming this got written like this due to a line-length rule? When the queue is less busy, could we revisit the topic of excluding this rule? I find it so disruptive to visually parsing the code when this happens. And it hurts my feelings. 😭

File: gp-inventory/gpi-waiting-list.php [hunk 2]
```diff
@@ -8,99 +8,197 @@
...
+		), 10, 2 );
+
+		add_filter( 'gform_entries_field_value', function ( $value, $form_id, $field_id, $entry ) {
```
Comment (spivurno) [inline gp-inventory/gpi-waiting-list.php:35]: Not a deal breaker but it does feel a little strange to be using anonymous functions now that this is a class?

**Review by spivurno — APPROVED** [review]
Awesome, man! I probably should have converted this to a class on my last passthrough.

Approved with some comments inline.

File: gp-inventory/gpi-waiting-list.php (hunk 1 above)
Comment (claygriffiths) [reply gp-inventory/gpi-waiting-list.php:32]: I think this was PhpStorm reformatting. Fixing

**Pull request merged**